README.md

```diff
@@ -1,15 +1,314 @@
+transactions:
+
+```go
+type RevealRecord struct {
+	DecryptionKey      []byte
+	TransactionRecords []RevealedTransactionRecord
+}
+
+type RevealedTransactionRecord struct {
+	Status            uint64
+	CumulativeGasUsed uint64
+	CumulativeLogs    uint64
+}
+
+const (
+	RevealedTransactionRecordStatusSuccess          = 0x64
+	RevealedTransactionRecordStatusDecryptionFailed = 0x65
+	RevealedTransactionRecordStatusExecutionFailed  = 0x66
+)
+```
+
+By definition, the `RevealRecord` will be in log position `0` of the shutterized block.
+
+### Usage
+
+Usage is very similar to that of a vanilla `ethers` `BrowserProvider`. If you're not familiar with
+it yet, you should [read the documentation](https://docs.ethers.org/v5/api/providers/).
+
+The following is an annotated version of the example in `example/browserwallet.ts`.
+
+#### Import
+
+```js
+import {
+  ShutterProvider,
+  ethers,
+  init,
+  decrypt,
+} from '@shutter-network/shop-sdk'
+```
+
+#### Chain specific parameters
+
+```js
+// shutter specific constants
+const options = {
+  // path to the WASM library
+  wasmUrl: '/shutter-crypto.wasm',
+  // contract address to query the current keyper set
+  keyperSetManagerAddress: '0x4200000000000000000000000000000000000067',
+  // target address for encrypted transactions
+  inboxAddress: '0x4200000000000000000000000000000000000066',
+  // contract address to query encryption keys
+  keyBroadcastAddress: '0x4200000000000000000000000000000000000068',
+}
+```
+
+In the case of Shutter Optimism, the contract addresses are "predeployed" at fixed addresses. So
+you don't have to expect these to change.
+
+#### Initialize WASM
+
+In order for the WebAssembly to become available, it needs to be initialized. This is achieved by
+the `init()` method:
+
+```js
+// initialize shutter WASM library
+await init(options.wasmUrl)
+```
+
+Note, that the `.wasm` file needs to be accessible for the browser, so usually it is a good idea
+to bundle it with other `assets` of your application.
+
+#### Provider / Signer
+
+The following is almost a copy & paste of the
+[original ethers documentation](https://docs.ethers.org/v6/getting-started/#starting-connecting).
+Pay attention to the creation of the `ShutterProvider` - you need to pass the `options` created
+earlier.
+
+```js
+let signer = null
+
+let provider
+if (window.ethereum == null) {
+  // If MetaMask is not installed, we use the default provider,
+  // which is backed by a variety of third-party services (such
+  // as INFURA). They do not have private keys installed,
+  // so they only have read-only access
+  console.log('MetaMask not installed; using read-only defaults')
+  provider = ethers.getDefaultProvider()
+} else {
+  // Connect to the MetaMask EIP-1193 object. This is a standard
+  // protocol that allows Ethers access to make all read-only
+  // requests through MetaMask.
+  // SHUTTER> use ShutterProvider
+  provider = new ShutterProvider(options, window.ethereum)
```

Comment:
Question: Is it possible to create a ShutterProvider instance by using an RPC url?

Reply:
I don't think so - `ShutterProvider` [extends `BrowserProvider`](https://github.com/ethers-io/ethers.js/blob/v6.12.1/src.ts/providers/provider-browser.ts#L44), so you'd need to wrap your RPC url in an implementation of an [`Eip1193Provider`](https://github.com/ethers-io/ethers.js/blob/v6.12.1/src.ts/providers/provider-browser.ts#L16) (usually provided by the browser plugin). I guess it is potentially doable, but I don't know how much effort it would be.